components/led_strip/led_vu.c

```diff
@@ -70,31 +72,31 @@ static void battery_svc(float value, int cells) {
 	if (battery_handler_chain) battery_handler_chain(value, cells);
 }
 
+/****************************************************************************************
+ * Suspend.
+ * 
+ */
+static void led_vu_sleep(void) {
+    led_vu_clear(led_display); 
+}
+
 /****************************************************************************************
  * Initialize the led vu strip if configured.
  * 
  */
 void led_vu_init()
 {
-    char* p;
     char* config = config_alloc_get_str("led_vu_config", NULL, "N/A");
 
-    // Initialize led VU strip 
-    char* drivername = strcasestr(config, "WS2812");
-
-    if ((p = strcasestr(config, "length")) != NULL) {
-        strip.length = atoi(strchr(p, '=') + 1);
-    } // else 0
-    if ((p = strcasestr(config, "gpio")) != NULL) {
-        strip.gpio = atoi(strchr(p, '=') + 1);
-    } else {
-        strip.gpio = LED_VU_DEFAULT_GPIO;
-    }
+    PARSE_PARAM(config, "length",'=', strip.length);
+    PARSE_PARAM(config, "gpio",'=', strip.gpio);
     // check for valid configuration
-    if (!drivername || !strip.gpio) {
+    if (!strip.gpio) {
```

Comment:
I'll take the PR but although I'm not at all a dogmatic anti-goto, I think they make sense when you have a series of cleanup actions that you don't want to repeat all the time. Here, because you have a one-line return that you "go to", I think a return instead of the goto would be better. It's just an opinion and that part is your code, so as long as it does not hurt 😄, it's fine.

Reply:
Thanks, will look at it